Libs/vtkSegmentationCore/vtkBinaryLabelmapToClosedSurfaceConversionRule.cxx

```diff
@@ -288,24 +294,81 @@ bool vtkBinaryLabelmapToClosedSurfaceConversionRule::CreateClosedSurface(vtkOrie
   double smoothingFactor = this->ConversionParameters->GetValueAsDouble(GetSmoothingFactorParameterName());
   int computeSurfaceNormals = this->ConversionParameters->GetValueAsInt(GetComputeSurfaceNormalsParameterName());
 
-  vtkNew<vtkDiscreteFlyingEdges3D> marchingCubes;
-  marchingCubes->SetInputData(binaryLabelmapWithIdentityGeometry);
-  marchingCubes->ComputeGradientsOff();
-  marchingCubes->ComputeNormalsOff(); // While computing normals is faster using the flying edges filter,
-  // it results in incorrect normals in meshes from shared labelmaps marchingCubes->ComputeScalarsOn();
+  // Generation method
+  // 0 = flying edges
+  // 1 = surface nets
+  int generationMethod = this->ConversionParameters->GetValueAsInt(GetConversionMethodParameterName());
+  int surfaceNetsSmoothing = this->ConversionParameters->GetValueAsInt(GetInternalSmoothingParameterName());
```

Comment:
`InternalSmoothingParameterName` -> `SurfaceNetInternalSmoothingParameterName`

And rename ivar accordingly